src/main/kotlin/com/cognifide/gradle/aem/environment/EnvironmentOptions.kt

```diff
@@ -0,0 +1,45 @@
+package com.cognifide.gradle.aem.environment
+
+import com.cognifide.gradle.aem.common.Retry
+import com.cognifide.gradle.aem.environment.docker.DockerOptions
+import com.cognifide.gradle.aem.environment.hosts.HostsOptions
+import java.net.URI
+import kotlin.math.max
+
+class EnvironmentOptions {
+    val healthChecks = HealthChecks()
+    val hosts = HostsOptions()
+    val docker = DockerOptions()
+
+    fun healthChecks(configurer: HealthChecks.() -> Unit) {
+        healthChecks.apply(configurer)
+    }
+
+    fun hosts(config: Map<String, String>) {
+        hosts.configure(config)
+    }
+
+    fun docker(configurer: DockerOptions.() -> Unit) {
+        docker.apply(configurer)
+    }
+}
+
+class HealthChecks {
+    val list = mutableListOf<HealthCheck>()
+
+    infix fun String.respondsWith(block: HealthCheck.() -> Unit) {
+        list += HealthCheck(this).apply(block)
+    }
+}
+
+class HealthCheck(val url: String) {
+    val uri = URI(url)
+    var status = 200
+    var text: String = ""
+    var maxAwaitTime = 60000
+    var connectionTimeout = 3000
+    var delay = Retry.SECOND_MILIS
+
+    fun numberOfChecks() = max(maxAwaitTime / connectionTimeout, 1)
```

Comment:
there is already `ProgressCountdown`, `Retry` etc. Reuse or add such generic purpose class and use it here.

Reply:
I changed the approach since I had problems to get it working correctly in this form. 
We can discuss how to integrate this with existing GAP code, for now, it is something separate:
```kotlin
fun ProgressLogger.awaitCondition(message: String, maxAwaitTime: Long, condition: suspend () -> Boolean): Boolean {
    var isValid = false
    runBlocking {
        withTimeoutOrNull(maxAwaitTime) {
            val startTime = System.currentTimeMillis()
            while (true) {
                val timePassed = System.currentTimeMillis() - startTime
                val remainingTime = (maxAwaitTime - timePassed) / Retry.SECOND_MILIS
                progress("!$remainingTime $message")
                delay(AWAIT_DELAY_DEFAULT)
                isValid = condition()
                if (isValid) {
                    return@withTimeoutOrNull
                }
            }
        }
    }
    return isValid
}
```